sql/inspect: TestInspectProgressWithMultiRangeTable failed

sql/inspect.TestInspectProgressWithMultiRangeTable [failed](https://mesolite.cluster.engflow.com/invocations/default/fd8b9a0d-f1e5-434c-bb19-01364df6a7b7?testReportRun=9&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2luc3BlY3Q6aW5zcGVjdF90ZXN0) on master @ [b3b7d649a57df992d63df60d08c7c59203d33be2](https://github.com/cockroachdb/cockroach/commits/b3b7d649a57df992d63df60d08c7c59203d33be2):


```
=== RUN   TestInspectProgressWithMultiRangeTable
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestInspectProgressWithMultiRangeTable3608498243
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:183: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    inspect_resumer_test.go:360: condition failed to evaluate within 45s: from inspect_resumer_test.go:373: waiting for scatter to multiple nodes: ranges only on 1 node(s)
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestInspectProgressWithMultiRangeTable3608498243
--- FAIL: TestInspectProgressWithMultiRangeTable (64.08s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInspectProgressWithMultiRangeTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59536

--- a/pkg/sql/inspect/BUILD.bazel
+++ b/pkg/sql/inspect/BUILD.bazel
@@ -104,6 +104,7 @@ go_test(
         "//pkg/keys",
         "//pkg/kv",
         "//pkg/kv/kvserver/protectedts",
+        "//pkg/multitenant/tenantcapabilitiespb",
         "//pkg/roachpb",
         "//pkg/security/securityassets",
         "//pkg/security/securitytest",

--- a/pkg/sql/inspect/inspect_resumer_test.go
+++ b/pkg/sql/inspect/inspect_resumer_test.go
@@ -8,6 +8,7 @@ package inspect_test
 import (
 	"context"
 	gosql "database/sql"
+	"fmt"
 	"regexp"
 	"sync/atomic"
 	"testing"
@@ -18,6 +19,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/jobs/jobfrontier"
 	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts"
+	"github.com/cockroachdb/cockroach/pkg/multitenant/tenantcapabilitiespb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
@@ -353,31 +355,24 @@ func TestInspectProgressWithMultiRangeTable(t *testing.T) {
 	for i := 100; i <= 900; i += 100 {
 		runner.Exec(t, `ALTER TABLE multi_range_table SPLIT AT VALUES ($1)`, i)
 	}
-	runner.Exec(t, `ALTER TABLE multi_range_table SCATTER`)
-
-	// Wait for scatter to distribute ranges across nodes.
-	var rangeCount, nodeCount int
-	testutils.SucceedsSoon(t, func() error {
-		// Count total ranges and verify distribution.
-		runner.QueryRow(t, `
-			WITH r AS (SHOW RANGES FROM TABLE multi_range_table WITH DETAILS)
-			SELECT count(DISTINCT range_id), count(DISTINCT lease_holder)
-			FROM r
-		`).Scan(&rangeCount, &nodeCount)
-
-		if rangeCount <= 5 {
-			return errors.Newf("waiting for splits: only %d ranges", rangeCount)
-		}
-
-		if nodeCount < 2 {
-			return errors.Newf("waiting for scatter to multiple nodes: ranges only on %d node(s)", nodeCount)
-		}
-
-		return nil
-	})
+	if tc.DefaultTenantDeploymentMode().IsExternal() {
+		tc.GrantTenantCapabilities(ctx, t, serverutils.TestTenantID(),
+			map[tenantcapabilitiespb.ID]string{tenantcapabilitiespb.CanAdminRelocateRange: "true"})
+	}
+	splitVals := []int{100, 200, 300, 400, 500, 600, 700, 800, 900}
+	for i, splitVal := range splitVals {
+		nodeIdx := (i % numNodes) + 1
+		stmt := fmt.Sprintf(
+			`ALTER TABLE multi_range_table EXPERIMENTAL_RELOCATE LEASE VALUES (%d, %d)`,
+			nodeIdx, splitVal)
+		testutils.SucceedsSoon(t, func() error {
+			_, err := db.ExecContext(ctx, stmt)
+			return err
+		})
+	}
 
 	// Start the INSPECT job.
-	t.Logf("Starting INSPECT job on table with %d ranges distributed across %d nodes", rangeCount, nodeCount)
+	t.Log("Starting INSPECT job on multi-range table with leases distributed across nodes")
 	_, err := db.Exec(`
 		COMMIT;
 		INSPECT TABLE multi_range_table`)

